code/game/mecha/mech_fabricator.dm

```diff
@@ -76,22 +80,17 @@
 	materials.max_amount = (200000 + (T*50000))
 
 	// build time adjustment coefficient (1 -> 0.85 -> 0.7 -> 0.55)
-	T = 1.15
-	for(var/obj/item/stock_parts/micro_laser/Ma in component_parts)
-		T -= Ma.rating*0.15
-	time_coeff = T
+	for(var/obj/item/stock_parts/micro_laser/Ml in component_parts)
+		time_coeff -= Ml.rating*0.15
 
 	// resource cost adjustment coefficient (1 -> 0.8 -> 0.6 -> 0.4)
-	T = -1
-	for(var/obj/item/stock_parts/manipulator/Ml in component_parts)
-		T += Ml.rating
-	component_coeff = round(initial(time_coeff) - (T / 5), 0.01)
+	for(var/obj/item/stock_parts/manipulator/Ma in component_parts)
+		component_coeff -= Ma.rating*0.20
 
 /obj/machinery/mecha_part_fabricator/interact(mob/user)
 	ui_interact(user)
 
 /obj/machinery/mecha_part_fabricator/ui_interact(mob/user, ui_key = "main", datum/nanoui/ui = null, force_open = 1)
-	user.set_machine(src)

```

Comment:
Oh, I removed this because I though it did something else, but after some research, I actually need this. It ultimately tells the machine if someone is using it, and if it has to start processing/updating things.